File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int n = nums.length;
        if(n==1)return target==nums[0]?0:-1;

        for(int left=0, right=n-1;left<=right;){
          if(!(left > -1 && left<n && right > -1 && right < n)) return -1;
          int mid = (left+right)/2;
          int val = nums[mid];
          if(val==target){
            return mid;
          }else if(target < val){
            right = mid-1;
          }else{
            left = mid+1;
          }
        }
        return -1;
    }
}

File: 792-binary-search/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l,r = 0, len(nums)-1
        while l<=r:
            m = (l+r)//2
            if l==r:
                return m if target == nums[m] else -1
            if target == nums[m]:
                return m
            elif target < nums[m]:
                r = m-1
            else:
                l = m+1
        return -1